File: problem3.java
// The solution implements a HashMap and HashSetto check if a string follows a given pattern.
class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] strArray = s.split(" ");
        if(pattern.length() != strArray.length) {
            return false;
        }
        HashMap<Character, String> patternMap = new HashMap<>();
        HashSet<String> set = new HashSet<>();
        for(int i=0; i<pattern.length(); i++) {
            Character c = pattern.charAt(i);
            String str = strArray[i];
            if(patternMap.containsKey(c)) {
                if(!str.equals(patternMap.get(c))) {
                    return false;
                }
            }
            else {
                    patternMap.put(c, str);
                    if(set.contains(str)) {
                        return false;
                    }
                    set.add(str);
                }
        }
        return true; 
    }
}

File: problem3.py
#The solution implements a HashMap check if a string follows a given pattern.
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        strList = s.split(" ")
        if len(pattern) != len(strList):
            return False
        patternMap = {}
        strMap = {}
        for i in range(len(pattern)):
            c = pattern[i]
            word = strList[i]
            if c in patternMap:
                if patternMap.get(c) != word:
                    return False
            else:
                patternMap[c] = word
            if word in strMap:
                if strMap.get(word) != c:
                    return False
            else:
                strMap[word] = c
        return True